controller/src/main/java/io/pravega/controller/server/rpc/grpc/v1/ControllerServiceImpl.java

```diff
@@ -416,7 +486,7 @@ public void getURI(SegmentId request, StreamObserver<NodeUri> responseObserver)
         log.info("getURI called for segment {}/{}/{}.", request.getStreamInfo().getScope(),
                 request.getStreamInfo().getStream(), request.getSegmentId());
         authenticateExecuteAndProcessResults(() -> this.grpcAuthHelper.checkAuthorization(
-                authorizationResource.ofStreamInScope(request.getStreamInfo().getScope(), request.getStreamInfo().getStream()),
+                StreamAuthParams.toResourceString(request.getStreamInfo().getScope(), request.getStreamInfo().getStream()),
```

Comment:
This change makes sure that the resource string is adapted based on whether the stream is internal/external and the type of resource it represents. All that logic is encapsulated within `StreamAuthParams.toResourceString()` so that that auth-specific logic doesn't leak into the business logic contained in this code in the Controller Service implementation.